src/components/MultiHopTrade/hooks/useAllowanceApproval/helpers.ts

```diff
@@ -63,9 +63,16 @@ export const getApprovalTxData = async (
     to: assetReference,
   })
 
-  const { networkFeeCryptoBaseUnit, ...fees } = await getFees({
-    accountNumber: tradeQuoteStep.accountNumber,
+  const from = await adapter.getAddress({
     wallet,
+    accountNumber: tradeQuoteStep.accountNumber,
```

Comment:
danger

Reply:
outside scope of this PR, but as discussed given sell asset account number through n-1 hops is the same, and only the buy or receive asset account number is different perhaps we should use that terminology again?